dotCMS/src/main/java/com/dotmarketing/cms/urlmap/filters/URLMapFilter.java

```diff
@@ -245,7 +250,12 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
                         if (optSession != null) {
                             optSession.setAttribute(com.dotmarketing.util.WebKeys.HTMLPAGE_LANGUAGE,
                                     String.valueOf(
-                                            conAPI.find(c.getInode(), user, true).getLanguageId()));
+                                            conAPI.find(c.getInode(), wuserAPI.getSystemUser(), true).getLanguageId()));
+                        }
+                        Contentlet con = conAPI.find(c.getInode(), wuserAPI.getSystemUser(), true);
+                        if(!perAPI.doesUserHavePermission(con, PermissionBitAPIImpl.PERMISSION_READ,user,true)){
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)